File: das_backend/src/main/java/ch/sbb/backend/logging/domain/service/DomainTenantService.java
package ch.sbb.backend.logging.domain.service;

import ch.sbb.backend.logging.domain.model.Tenant;
import ch.sbb.backend.logging.domain.repository.TenantRepository;

public class DomainTenantService implements TenantService {

    private final TenantRepository tenantRepository;

    public DomainTenantService(TenantRepository tenantRepository) {
        this.tenantRepository = tenantRepository;
    }

    @Override
    public Tenant getByIssuerUri(String issuerUri) {
        return tenantRepository.getByIssuerUri(issuerUri);
    }

    @Override
    public Tenant current() {
        return tenantRepository.current();
    }
}

File: das_backend/src/main/kotlin/ch/sbb/backend/logging/domain/service/DomainTenantService.kt
package ch.sbb.backend.logging.domain.service

import ch.sbb.backend.logging.domain.Tenant
import ch.sbb.backend.logging.domain.repository.TenantRepository

class DomainTenantService(private val tenantRepository: TenantRepository) : TenantService {
    override fun getByIssuerUri(issuerUri: String): Tenant {
        return tenantRepository.getByIssuerUri(issuerUri)
    }

    override fun current(): Tenant {
        return tenantRepository.current()
    }
}
